GitCommands/Settings/AppSettings.cs

```diff
@@ -1179,13 +1179,11 @@ public static void SaveSettings()
         public static void LoadSettings()
         {
             Action<Encoding> addEncoding = delegate(Encoding e) { AvailableEncodings[e.HeaderName] = e; };
-            addEncoding(Encoding.Default);
-            addEncoding(new ASCIIEncoding());
-            addEncoding(new UnicodeEncoding());
-            addEncoding(new UTF7Encoding());
-            addEncoding(new UTF8Encoding(false));
-            addEncoding(System.Text.Encoding.GetEncoding("CP852"));
-
+            foreach(var enc in Encoding.GetEncodings())
+            {
+                addEncoding(enc.GetEncoding());
+            }
```

Comment:
what is the use-case to have all of them?

I don't think it is a good idea to overload the user with the unnecessary data. Would it perhaps be better to allow the user to toggle all encoding, if some quirky encoding is required? 
And by default the user would continue to see the selected list of encodings.